include/taco/linalg_notation/linalg_rewriter.h

```diff
@@ -0,0 +1,75 @@
+#ifndef TACO_LINALG_REWRITER_H
+#define TACO_LINALG_REWRITER_H
+
+#include <vector>
+#include <map>
+#include <set>
+#include <memory>
+#include <taco/index_notation/index_notation.h>
+
+#include "taco/lower/iterator.h"
+#include "taco/util/scopedset.h"
+#include "taco/util/uncopyable.h"
+#include "taco/ir_tags.h"
+
+namespace taco {
+
+class TensorVar;
+
+class IndexVar;
+
+class IndexExpr;
+
+class LinalgBase;
+
+class LinalgRewriter : public util::Uncopyable {
```

Comment:
The name "rewriter" is somewhat confusing since it suggests that `rewrite` returns a new `LinalgStmt` or `LinalgExpr` as input (similar to how `IndexNotationRewriter` takes `IndexExpr` as input and returns `IndexExpr` as output). Maybe something like `LinalgLowerer` would be less confusing?